api/api.go

```diff
@@ -375,6 +375,9 @@ func procDeleteNetwork(c libnetwork.NetworkController, vars map[string]string, b
 	if !errRsp.isOK() {
 		return nil, errRsp
 	}
+	if isReservedNetwork(nw.Name()) {
```

Comment:
@aboch @nerdalert In fact my suggestion is not even do it libnetwork but do it instead in the driver. And simply check with the driver before you delete a network object. For `null` and `host` it's easy. Always return an error for DeleteNetwork api in the driver. For `bridge` do not delete it if the `bridgeName` is `DefaultBridgeName`. And that's it. No special casing at all in the libnetwork code.